dashboard/cron.py

```diff
@@ -246,18 +248,9 @@ def update_with_bq_access(self):
         # BQ Total Bytes Billed to report to status
         total_bytes_billed = 0
 
-        # the earliest start date of all courses
-        course_start_time = utils.find_earliest_start_datetime_of_courses()
-
-        # the earliest latest date of all resources_accessed
-        try:
-            latest_resource = ResourceAccess.objects.latest('access_time')
-            latest_resource_time = latest_resource.access_time
-        except ResourceAccess.DoesNotExist:
-            latest_resource_time = course_start_time
-
-        logger.info(f"Earliest Start: {course_start_time} Latest resource: {latest_resource_time}")
+        last_cron_run = Course.objects.last_cron_run()
 
+        status += deleteAllRecordInTable("resource_access", f"WHERE access_time > %s", [last_cron_run,])
```

Comment:
Feels like there is a reason for this, but having trouble articulating it specifically. Can you explain? If `last_cron_run` uses a course start date, it'd reset `resource_access` essentially. If it's the last time the cron ran, it'd wipe out records after that, but would there be any?

Reply:
Yeah my idea is it last_cron_run (now renamed to data_last_updated) would return the start date of the new courses. This is either going to be the date set on the course, the start of the term or "now" if all those fail. (That logic is in course_date_range)

So there's a chance it will wipe it all out. However if we have multiple terms (say Winter 2019 and Winter 2020) and the new course is just Winter 2020, it will just wipe out data up to Winter 2020 and repopulate that. This is really just protecting from a full database wipeout for really old courses. Maybe that isn't as necessary as I'm thinking.

Reply:
Oh, I was thinking that the `get_last_cron_run` was acting on all courses, not just the new ones. This is starting to make more sense. So is this mainly to reduce duplicates when adding new data? `pangres` upsert would handle them, but you're making things easier?

Reply:
Yeah, this is really just to set the *best* date for the BigQuery date. Pangres is fine for updating resources but we don't have any easy way to upsert into the resources_access, so it has to delete up to a date and re-insert.

Reply:
Oh, gotcha, thought the upsert was happening here, too. I'm all caught up. This should be fine.